adr/0004-supported-python-versions.md

```diff
@@ -0,0 +1,36 @@
+# Supported Python Versions
+
+Date: 2017-11-20
+
+## Status
+
+Proposed
+
+## Context 
+
+eLife has numerous projects written completely and partly with the [Python programming language](https://www.python.org/).
+
+In order to provide language version consistency across projects we need to get a consensus on which versions we are going to support. 
+
+## Descision
+
+We will use Python 2.7.14 as our default version for any project that solely uses or supports Python 2.
+
+We will use Python 3.6.3 as our default supported version for any project that solely uses or supports Python 3.
+
+## Consequences
+
+Remote environments that come with Python 3.5 as their default Python 3 interpreter and that can't easily install Python 3.6 may not be able to adhere to this.
+
+Projects that support Python 3.6 will gain from the numerous benefits of the release:
```

Comment:
Libraries (that are used in 2.x-only projects) may have to support both major versions? For how long?

Reply:
Good questions. 

I think we need to come up with a list of projects that will never need Python 3 support and why? presumably if we never plan to upgrade the project to Python 3 then the project is going to be replaced or archived in the near future? 

For brand new projects moving forward a descission can be made at the time whether to offer support for both, libraries probably require both depending on use case, stand alone apps etc could just use Python 3.

Reply:
There's at least one ``elife-poa-xml-generation`` I would not bother supporting Python 3, depending on when we have replacement libraries. Anything we no longer use and we decide to archive I don't see the benefit in upgrading.